bundles/org.openhab.binding.nuki/src/main/java/org/openhab/binding/nuki/internal/handler/AbstractNukiDeviceHandler.java

```diff
@@ -0,0 +1,320 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.nuki.internal.handler;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.nuki.internal.constants.NukiBindingConstants;
+import org.openhab.binding.nuki.internal.dataexchange.BridgeListResponse;
+import org.openhab.binding.nuki.internal.dataexchange.BridgeLockStateResponse;
+import org.openhab.binding.nuki.internal.dataexchange.NukiBaseResponse;
+import org.openhab.binding.nuki.internal.dataexchange.NukiHttpClient;
+import org.openhab.binding.nuki.internal.dto.BridgeApiDeviceStateDto;
+import org.openhab.binding.nuki.internal.dto.BridgeApiListDeviceDto;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingStatusInfo;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.BridgeHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AbstractNukiDeviceHandler} is a base class for implementing ThingHandlers for Nuki devices
+ *
+ * @author Jan Vybíral - Initial contribution
+ */
+@NonNullByDefault
+public abstract class AbstractNukiDeviceHandler extends BaseThingHandler {
+
+    protected final Logger logger = LoggerFactory.getLogger(getClass());
+    private static final int JOB_INTERVAL = 60;
+
+    @Nullable
+    private NukiHttpClient nukiHttpClient;
+    @Nullable
+    protected ScheduledFuture<?> reInitJob;
+    protected final String nukiId;
+
+    public AbstractNukiDeviceHandler(Thing thing) {
+        super(thing);
+        logger.debug("Instantiating {}({})", getClass().getSimpleName(), thing);
```

Comment:
Can this message be replaced by using the debugger? See point 4 https://www.openhab.org/docs/developer/guidelines.html#f-logging
```
void myFun() {
    logger.trace("Enter myfun"); // DONT, DONT, really DONT do that
    doSomething();
    logger.trace("Leave myfun"); // DONT, DONT, really DONT do that
}
```

Please check all.